### PR TITLE
covscript: Add version 3.3.2

### PR DESCRIPTION
**Pull request by issaclin32**
**covscript** (Covariant Script) ([Github repo](https://github.com/covscript/covscript)) is a scripting language.

Notes:
* **checkver**: This manifest matches new version from `https://github.com/covscript/covscript/releases` instead of **latest release** because some versions (e.g. 3.3.1, 3.3.3) does not provide pre-compiled binaries.

### CODE REVIEW

File: bucket/covscript.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            "extract_dir": "build_x64"
+        },
+        "32bit": {
+            "url": "https://github.com/covscript/covscript/releases/download/3.3.2/covscript-win32.7z",
+            "hash": "ddc319251260bc3aab35c931c6e7761a1d8499839fb9965655653999c3de0f27",
+            "extract_dir": "build"
+        }
+    },
+    "pre_install": [
+        "Set-Content \"$dir\\bin\\csbuild.ps1\" 'if ($args) { $a=$args } else { $a=@() }; Start-Process \"$PSScriptRoot\\csbuild.bat\" $a' -Encoding Ascii",
```
Comment (Ash258) [inline bucket/covscript.json]: Why do you do some argument check??

File: bucket/covscript.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            "hash": "c1b250bbc995982c33f1e33fefd62a794626d131e71c116145b23a1d2bd076f8",
+            "extract_dir": "build_x64"
+        },
+        "32bit": {
+            "url": "https://github.com/covscript/covscript/releases/download/3.3.2/covscript-win32.7z",
+            "hash": "ddc319251260bc3aab35c931c6e7761a1d8499839fb9965655653999c3de0f27",
+            "extract_dir": "build"
+        }
+    },
+    "pre_install": [
```
Comment (Ash258) [inline bucket/covscript.json]: Why do you do this? What is wrong with .bat shims?

A

Clean .bat shim:

image

File: bucket/covscript.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
+{
+    "version": "3.3.2",
+    "description": "Covariant Script interpreter",
+    "homepage": "https://covscript.org/",
```
Comment (Ash258) [inline bucket/covscript.json]:
```suggestion
    "homepage": "https://covscript.org",
```

File: bucket/covscript.json (hunk 2 above)
Comment (issaclin32) [reply bucket/covscript.json]: I was wrong. 
*.bat* shims works just fine. Therefore the ps1 script is not needed.

File: bucket/covscript.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        "32bit": {
+            "url": "https://github.com/covscript/covscript/releases/download/3.3.2/covscript-win32.7z",
+            "hash": "ddc319251260bc3aab35c931c6e7761a1d8499839fb9965655653999c3de0f27",
+            "extract_dir": "build"
+        }
+    },
+    "bin": [
+        "bin\\cs.exe",
+        "bin\\cs_dbg.exe",
+        "bin\\csbuild.bat",
```
Comment (Ash258) [inline bucket/covscript.json]: Is this correct? Should it be used in some specific folder?

A

Comment (issaclin32) [reply bucket/covscript.json]: Yes. This is correct. It should be considered an "upstream issue".
Sorry that I did not mention this in the comments.

Comment (Ash258) [reply bucket/covscript.json]: Do you have issue id or link for it?

**Comment by issaclin32** [comment]
I am closing this PR because
(1) It is outdated.
(2) I have not get the response (from the developer of *covscript*) regarding to the issue mentioned above.

**Pull request closed without merging**